massa-execution-worker/src/speculative_ledger.rs

```diff
@@ -127,7 +143,7 @@ impl SpeculativeLedger {
         if let Some(from_addr) = from_addr {
```

Comment:
To accomodate for the transfer_coins_for ABI, we also need to have sender_address here. But it needs to be optional because it is absen tin some cases (eg. block rewards credit)

Reply:
hmmm. actually this one is complicated. Forget about it for now :)

Reply:
Can you open an issue to discuss it as a followup ?